bin/network-monitor/src/remote_prover.rs

```diff
@@ -0,0 +1,313 @@
+            ProofType::Block => proto::remote_prover::ProofType::Block,
+            ProofType::Batch => proto::remote_prover::ProofType::Batch,
+        }
+    }
+}
+
+impl From<proto::remote_prover::ProofType> for ProofType {
+    fn from(value: proto::remote_prover::ProofType) -> Self {
+        match value {
+            proto::remote_prover::ProofType::Transaction => ProofType::Transaction,
+            proto::remote_prover::ProofType::Batch => ProofType::Batch,
+            proto::remote_prover::ProofType::Block => ProofType::Block,
+        }
+    }
+}
+
+// REMOTE PROVER TEST TYPES
+// ================================================================================================
+
+/// Details of a remote transaction prover test.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ProverTestDetails {
+    pub test_duration_ms: u64,
+    pub proof_size_bytes: usize,
+    pub success_count: u64,
+    pub failure_count: u64,
+    pub proof_type: ProofType,
+}
+
+// REMOTE TRANSACTION PROVER TEST TASK
+// ================================================================================================
+
+/// Runs a task that continuously tests remote prover functionality and updates a watch channel.
+///
+/// This function spawns a task that periodically sends mock request payloads to a remote prover
+/// and measures the success/failure rate and performance metrics for proof generation.
+///
+/// # Arguments
+///
+/// * `prover_url` - The URL of the remote prover service to test.
+/// * `name` - The name of the remote prover.
+/// * `proof_type` - The type of proof to test.
+/// * `serialized_request_payload` - The serialized request payload to send to the remote prover.
+/// * `status_sender` - The sender for the watch channel.
+///
+/// # Returns
+///
+/// `Ok(())` if the task completes successfully, or an error if the task fails.
+#[instrument(target = COMPONENT, name = "remote-prover-test-task", skip_all)]
+pub async fn run_remote_prover_test_task(
+    prover_url: Url,
+    name: &str,
+    proof_type: ProofType,
+    serialized_request_payload: proto::remote_prover::ProofRequest,
+    status_sender: watch::Sender<ServiceStatus>,
+) -> anyhow::Result<()> {
+    let mut client = ClientBuilder::new(prover_url)
+        .with_tls()
+        .expect("TLS is enabled")
+        .with_timeout(Duration::from_secs(10))
+        .without_metadata_version()
+        .without_metadata_genesis()
+        .connect_lazy::<RemoteProverClient>();
+
+    let mut interval = tokio::time::interval(Duration::from_secs(2 * 60)); // Test every 2 minutes
+    interval.set_missed_tick_behavior(MissedTickBehavior::Skip);
+
+    let mut success_count = 0u64;
+    let mut failure_count = 0u64;
+
+    loop {
+        interval.tick().await;
+
+        let current_time = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .context("failed to get current time")?
+            .as_secs();
```

Comment:
I wonder if only doing `SystemTime::now()` here, and leaving the unix timestamp conversion to the `frontend.rs` handler call might make sense? That makes all of these functions infallible.

Reply:
Done! Did this change + replace the return type of the return function.